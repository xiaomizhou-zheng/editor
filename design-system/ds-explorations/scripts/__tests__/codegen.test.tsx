import { createColorStylesFromTemplate } from '../color-codegen-template';
import { createInteractionStylesFromTemplate } from '../interaction-codegen';
describe('@atlaskit/design-system-explorations', () => {
  test('text styles are generated correctly', () => {
    expect(createColorStylesFromTemplate('text')).toMatchSnapshot();
  });

  test('bg styles are generated correctly', () => {
    expect(createColorStylesFromTemplate('background')).toMatchSnapshot();
  });

  test('border styles are generated correctly', () => {
    expect(createColorStylesFromTemplate('border')).toMatchSnapshot();
  });

  test('border styles are generated correctly', () => {
    expect(createInteractionStylesFromTemplate('background')).toMatchSnapshot();
  });

  test('incorrect config throws', () => {
    expect(() =>
      createColorStylesFromTemplate('fizzbuzz' as any),
    ).toThrowError();
  });
});
