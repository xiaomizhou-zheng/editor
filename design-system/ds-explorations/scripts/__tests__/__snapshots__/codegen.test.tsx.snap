// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@atlaskit/design-system-explorations bg styles are generated correctly 1`] = `
"const backgroundColorMap = {
  disabled: css({
    backgroundColor: token('color.background.disabled', '#091e4289'),
  }),
  'inverse.subtle': css({
    backgroundColor: token('color.background.inverse.subtle', '#00000029'),
  }),
  input: css({
    backgroundColor: token('color.background.input', '#FAFBFC'),
  }),
  neutral: css({
    backgroundColor: token('color.background.neutral', '#DFE1E6'),
  }),
  'neutral.subtle': css({
    backgroundColor: token('color.background.neutral.subtle', 'transparent'),
  }),
  'neutral.bold': css({
    backgroundColor: token('color.background.neutral.bold', '#42526E'),
  }),
  'brand.bold': css({
    backgroundColor: token('color.background.brand.bold', '#0052CC'),
  }),
  selected: css({
    backgroundColor: token('color.background.selected', '#DEEBFF'),
  }),
  'selected.bold': css({
    backgroundColor: token('color.background.selected.bold', '#0052CC'),
  }),
  danger: css({
    backgroundColor: token('color.background.danger', '#FFEBE6'),
  }),
  'danger.bold': css({
    backgroundColor: token('color.background.danger.bold', '#DE350B'),
  }),
  warning: css({
    backgroundColor: token('color.background.warning', '#FFFAE6'),
  }),
  'warning.bold': css({
    backgroundColor: token('color.background.warning.bold', '#FFAB00'),
  }),
  success: css({
    backgroundColor: token('color.background.success', '#E3FCEF'),
  }),
  'success.bold': css({
    backgroundColor: token('color.background.success.bold', '#00875A'),
  }),
  discovery: css({
    backgroundColor: token('color.background.discovery', '#EAE6FF'),
  }),
  'discovery.bold': css({
    backgroundColor: token('color.background.discovery.bold', '#5243AA'),
  }),
  information: css({
    backgroundColor: token('color.background.information', '#DEEBFF'),
  }),
  'information.bold': css({
    backgroundColor: token('color.background.information.bold', '#0052CC'),
  }),
  'color.blanket': css({
    backgroundColor: token('color.blanket', '#091e4289'),
  }),
  'color.blanket.selected': css({
    backgroundColor: token('color.blanket.selected', '#388BFF14'),
  }),
  'color.blanket.danger': css({
    backgroundColor: token('color.blanket.danger', '#EF5C4814'),
  }),
  'elevation.surface': css({
    backgroundColor: token('elevation.surface', '#FFFFFF'),
  }),
  'elevation.surface.sunken': css({
    backgroundColor: token('elevation.surface.sunken', '#F4F5F7'),
  }),
  'elevation.surface.raised': css({
    backgroundColor: token('elevation.surface.raised', '#FFFFFF'),
  }),
  'elevation.surface.overlay': css({
    backgroundColor: token('elevation.surface.overlay', '#FFFFFF'),
  }),
};

export type BackgroundColor = keyof typeof backgroundColorMap;
"
`;

exports[`@atlaskit/design-system-explorations border styles are generated correctly 1`] = `
"const borderColorMap = {
  'color.border': css({
    borderColor: token('color.border', '#091e4221'),
  }),
  bold: css({
    borderColor: token('color.border.bold', '#344563'),
  }),
  inverse: css({
    borderColor: token('color.border.inverse', '#FFFFFF'),
  }),
  focused: css({
    borderColor: token('color.border.focused', '#4C9AFF'),
  }),
  input: css({
    borderColor: token('color.border.input', '#FAFBFC'),
  }),
  disabled: css({
    borderColor: token('color.border.disabled', '#FAFBFC'),
  }),
  brand: css({
    borderColor: token('color.border.brand', '#0052CC'),
  }),
  selected: css({
    borderColor: token('color.border.selected', '#0052CC'),
  }),
  danger: css({
    borderColor: token('color.border.danger', '#FF5630'),
  }),
  warning: css({
    borderColor: token('color.border.warning', '#FFC400'),
  }),
  success: css({
    borderColor: token('color.border.success', '#00875A'),
  }),
  discovery: css({
    borderColor: token('color.border.discovery', '#998DD9'),
  }),
  information: css({
    borderColor: token('color.border.information', '#0065FF'),
  }),
};

export type BorderColor = keyof typeof borderColorMap;
"
`;

exports[`@atlaskit/design-system-explorations border styles are generated correctly 2`] = `
"const backgroundActiveColorMap = {
  'inverse.subtle': css({
    ':active': {
      backgroundColor: token('color.background.inverse.subtle.pressed'),
    },
  }),
  input: css({
    ':active': { backgroundColor: token('color.background.input.pressed') },
  }),
  neutral: css({
    ':active': { backgroundColor: token('color.background.neutral.pressed') },
  }),
  'neutral.subtle': css({
    ':active': {
      backgroundColor: token('color.background.neutral.subtle.pressed'),
    },
  }),
  'neutral.bold': css({
    ':active': {
      backgroundColor: token('color.background.neutral.bold.pressed'),
    },
  }),
  'brand.bold': css({
    ':active': {
      backgroundColor: token('color.background.brand.bold.pressed'),
    },
  }),
  selected: css({
    ':active': { backgroundColor: token('color.background.selected.pressed') },
  }),
  'selected.bold': css({
    ':active': {
      backgroundColor: token('color.background.selected.bold.pressed'),
    },
  }),
  danger: css({
    ':active': { backgroundColor: token('color.background.danger.pressed') },
  }),
  'danger.bold': css({
    ':active': {
      backgroundColor: token('color.background.danger.bold.pressed'),
    },
  }),
  warning: css({
    ':active': { backgroundColor: token('color.background.warning.pressed') },
  }),
  'warning.bold': css({
    ':active': {
      backgroundColor: token('color.background.warning.bold.pressed'),
    },
  }),
  success: css({
    ':active': { backgroundColor: token('color.background.success.pressed') },
  }),
  'success.bold': css({
    ':active': {
      backgroundColor: token('color.background.success.bold.pressed'),
    },
  }),
  discovery: css({
    ':active': { backgroundColor: token('color.background.discovery.pressed') },
  }),
  'discovery.bold': css({
    ':active': {
      backgroundColor: token('color.background.discovery.bold.pressed'),
    },
  }),
  information: css({
    ':active': {
      backgroundColor: token('color.background.information.pressed'),
    },
  }),
  'information.bold': css({
    ':active': {
      backgroundColor: token('color.background.information.bold.pressed'),
    },
  }),
  'elevation.surface': css({
    ':active': { backgroundColor: token('elevation.surface.pressed') },
  }),
  'elevation.surface.raised': css({
    ':active': { backgroundColor: token('elevation.surface.raised.pressed') },
  }),
  'elevation.surface.overlay': css({
    ':active': { backgroundColor: token('elevation.surface.overlay.pressed') },
  }),
};

const backgroundHoverColorMap = {
  'inverse.subtle': css({
    ':hover': {
      backgroundColor: token('color.background.inverse.subtle.hovered'),
    },
  }),
  input: css({
    ':hover': { backgroundColor: token('color.background.input.hovered') },
  }),
  neutral: css({
    ':hover': { backgroundColor: token('color.background.neutral.hovered') },
  }),
  'neutral.subtle': css({
    ':hover': {
      backgroundColor: token('color.background.neutral.subtle.hovered'),
    },
  }),
  'neutral.bold': css({
    ':hover': {
      backgroundColor: token('color.background.neutral.bold.hovered'),
    },
  }),
  'brand.bold': css({
    ':hover': { backgroundColor: token('color.background.brand.bold.hovered') },
  }),
  selected: css({
    ':hover': { backgroundColor: token('color.background.selected.hovered') },
  }),
  'selected.bold': css({
    ':hover': {
      backgroundColor: token('color.background.selected.bold.hovered'),
    },
  }),
  danger: css({
    ':hover': { backgroundColor: token('color.background.danger.hovered') },
  }),
  'danger.bold': css({
    ':hover': {
      backgroundColor: token('color.background.danger.bold.hovered'),
    },
  }),
  warning: css({
    ':hover': { backgroundColor: token('color.background.warning.hovered') },
  }),
  'warning.bold': css({
    ':hover': {
      backgroundColor: token('color.background.warning.bold.hovered'),
    },
  }),
  success: css({
    ':hover': { backgroundColor: token('color.background.success.hovered') },
  }),
  'success.bold': css({
    ':hover': {
      backgroundColor: token('color.background.success.bold.hovered'),
    },
  }),
  discovery: css({
    ':hover': { backgroundColor: token('color.background.discovery.hovered') },
  }),
  'discovery.bold': css({
    ':hover': {
      backgroundColor: token('color.background.discovery.bold.hovered'),
    },
  }),
  information: css({
    ':hover': {
      backgroundColor: token('color.background.information.hovered'),
    },
  }),
  'information.bold': css({
    ':hover': {
      backgroundColor: token('color.background.information.bold.hovered'),
    },
  }),
  'elevation.surface': css({
    ':hover': { backgroundColor: token('elevation.surface.hovered') },
  }),
  'elevation.surface.raised': css({
    ':hover': { backgroundColor: token('elevation.surface.raised.hovered') },
  }),
  'elevation.surface.overlay': css({
    ':hover': { backgroundColor: token('elevation.surface.overlay.hovered') },
  }),
};

type InteractionBackgroundColor = keyof typeof backgroundHoverColorMap;
"
`;

exports[`@atlaskit/design-system-explorations text styles are generated correctly 1`] = `
"const textColorMap = {
  'color.text': css({
    color: token('color.text', '#172B4D'),
  }),
  subtle: css({
    color: token('color.text.subtle', '#42526E'),
  }),
  subtlest: css({
    color: token('color.text.subtlest', '#7A869A'),
  }),
  disabled: css({
    color: token('color.text.disabled', '#A5ADBA'),
  }),
  inverse: css({
    color: token('color.text.inverse', '#FFFFFF'),
  }),
  brand: css({
    color: token('color.text.brand', '#0065FF'),
  }),
  selected: css({
    color: token('color.text.selected', '#0052CC'),
  }),
  danger: css({
    color: token('color.text.danger', '#DE350B'),
  }),
  warning: css({
    color: token('color.text.warning', '#974F0C'),
  }),
  'warning.inverse': css({
    color: token('color.text.warning.inverse', '#172B4D'),
  }),
  success: css({
    color: token('color.text.success', '#006644'),
  }),
  discovery: css({
    color: token('color.text.discovery', '#403294'),
  }),
  information: css({
    color: token('color.text.information', '#0052CC'),
  }),
};

export type TextColor = keyof typeof textColorMap;
"
`;
